catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/config/HadoopS3FileSystemConfig.java

```diff
@@ -0,0 +1,112 @@
+
+  public HadoopS3FileSystemConfig(Map<String, String> properties) {
+    super(false);
+    loadFromMap(properties, k -> true);
+  }
+
+  public static final ConfigEntry<String> HADOOP_S3_ENDPOINT_ENTRY =
+      new ConfigBuilder(S3Properties.GRAVITINO_S3_ENDPOINT)
+          .doc("The endpoint of the AWS s3")
+          .version(ConfigConstants.VERSION_0_7_0)
+          .stringConf()
+          .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
+          .create();
+
+  public static final ConfigEntry<String> HADOOP_S3_ACCESS_KEY_ENTRY =
+      new ConfigBuilder(S3Properties.GRAVITINO_S3_ACCESS_KEY_ID)
+          .doc("The access key of the AWS s3")
+          .version(ConfigConstants.VERSION_0_7_0)
+          .stringConf()
+          .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
+          .create();
+
+  public static final ConfigEntry<String> HADOOP_S3_SECRET_KEY_ENTRY =
+      new ConfigBuilder(S3Properties.GRAVITINO_S3_SECRET_ACCESS_KEY)
+          .doc("The secret key of the AWS s3")
+          .version(ConfigConstants.VERSION_0_7_0)
+          .stringConf()
+          .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
+          .create();
+
+  public static final ConfigEntry<String> HADOOP_S3_CREDENTIAL_PROVIDER_KEY_ENTRY =
+      new ConfigBuilder(S3Properties.GRAVITINO_S3_CREDS_PROVIDER)
+          .doc("The S3 credentials provider class name")
+          .version(ConfigConstants.VERSION_0_7_0)
+          .stringConf()
+          .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
+          .create();
+
+  public static final Map<String, PropertyEntry<?>> S3_FILESYSTEM_PROPERTY_ENTRIES =
+      new ImmutableMap.Builder<String, PropertyEntry<?>>()
+          .put(
+              S3Properties.GRAVITINO_S3_ENDPOINT,
+              PropertyEntry.stringOptionalPropertyEntry(
+                  S3Properties.GRAVITINO_S3_ENDPOINT,
+                  "The endpoint of the AWS s3",
+                  false /* immutable */,
+                  null /* defaultValue */,
+                  false /* hidden */))
+          .put(
+              S3Properties.GRAVITINO_S3_ACCESS_KEY_ID,
+              PropertyEntry.stringOptionalPropertyEntry(
+                  S3Properties.GRAVITINO_S3_ACCESS_KEY_ID,
+                  "The access key of the AWS s3",
+                  false /* immutable */,
+                  null /* defaultValue */,
+                  false /* hidden */))
+          .put(
+              S3Properties.GRAVITINO_S3_SECRET_ACCESS_KEY,
+              PropertyEntry.stringOptionalPropertyEntry(
+                  S3Properties.GRAVITINO_S3_SECRET_ACCESS_KEY,
+                  "The secret key of the AWS s3",
+                  false /* immutable */,
+                  null /* defaultValue */,
+                  false /* hidden */))
+          .put(
+              S3Properties.GRAVITINO_S3_CREDS_PROVIDER,
```

Comment:
Why S3 has this credential provider property, but gcs doesn't?

Reply:
Yes, Gcs only needs to specify the service JSON file.